Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -375,6 +381,31 @@ namespace AzToolsFramework
             return AZStd::move(patch);
         }
 
+        void PrefabPublicHandler::UpdateTemplateForInstance(const TemplateId templateId, const Instance& instanceToUpdate)
+        {
+            AZ_Assert(instanceToUpdate.GetTemplateId() == templateId, "PrefabPublicHandler::UpdateTemplateForInstance - "
+                "The given template id does not match the template id that the instance belongs to.");
+
+            // Before updating we reset the transform data, so the transform data will be excluded in template.
+            const AZ::EntityId containerEntityId = instanceToUpdate.GetContainerEntityId();
+
+            AZ::EntityId parentEntityIdBeforeReset;
+            AZ::TransformBus::EventResult(parentEntityIdBeforeReset, containerEntityId, &AZ::TransformBus::Events::GetParentId);
+
+            AZ::Transform localTransformBeforeReset = AZ::Transform::CreateIdentity();
+            AZ::TransformBus::EventResult(localTransformBeforeReset, containerEntityId, &AZ::TransformBus::Events::GetLocalTM);
+
+            AZ::TransformBus::Event(containerEntityId, &AZ::TransformBus::Events::SetParent, AZ::EntityId());
+            AZ::TransformBus::Event(containerEntityId, &AZ::TransformBus::Events::SetLocalTM, AZ::Transform::CreateIdentity());
```

Comment:
All of this is not needed because the parent entity and transform info is not set on the new entity at this point.

Reply:
New container entity object's transform is updated by ApplyContainerTransformAndGeneratePatch. The call was placed before setting top level entities' parent. It looks like it was placed there by purpose. Still couldn't figure out why.

I tried to delay calling ApplyContainerTransformAndGeneratePatch to one line before CreateLink, but some prefab tests fail due to transform position mismatch (not for the new container entity, but its child).

> assert current_entity_pos.IsClose(expected_pos), \\\n  AssertionError: Entity 'Entity_0' position '(x=200.0000000,y=200.0000000,z=200.0000000)' is not located at expected position '(x=100.0000000,y=100.0000000,z=100.0000000)'\nTest result:  FAILURE"}

Reply:
But you can avoid the resetting if you do the entity dom update this way.
- store the before state of entity in a dom by fetching it from the template just after the previous template update
- serialize the entity again in this function to get the after state
- generate a patch by comparing the doms
- apply that patch to the prefab template (with undo/redo support---because we need this info again if user does undo followed by redo)

I think this way of updating the template would be better because it captures all the diff of the container entity since the last template update and you don't have to do additional resetting logic here.

Reply:
Discussed offline: (1) The reset can be avoided if comparing two entity DOMs that both have the same transform info (even if it contains parent info). (2) The undo/redo is not needed.